recipes/botan/all/conanfile.py

```diff
@@ -217,6 +245,48 @@ def _configure_cmd(self):
         if self.settings.build_type == 'RelWithDebInfo':
             build_flags.append('--with-debug-info')
 
+        if not self.options.with_sse2:
+            build_flags.append('--disable-sse2')
```

Comment:
Isn't this non-deterministic?
This will depend on the actual build system used.
e.g. an older system without sse4 vs a newer version. Or amd cpu vs intel cpu.

Reply:
not really. compiler may generate SSE4 or newer code, even if build machine doesn't have it.

Reply:
Ok, but then it should at least do this because now it depends on the default configuration (which might change):
```
build_flags.append('--enable-sse2' if self.options.get_safe('with_sse2') else '--disable-sse2')
```

Reply:
yes and botan checks wether the current runtime environment has the desired isa extension.

 for example see the usage of "CPUID::has_avx2()" in botan.. they had a compile time check and runtime.

Reply:
I just want to avoid having the options of this recipe out of sync with the actual options botan ends up compiled with.

Reply:
I agree with @madebr

Reply:
I've added required checks